### PR TITLE
"Rewrite" section cleanup

### LINKED ISSUE #236: "rewrite" documentation has confusing headings
http://docs.couchdb.org/en/2.1.1/api/ddoc/rewrites.html#rewrite-section-a-is-stringified-function

This page says 'Rewrite section a is stringified function'. Is that supposed to be 'Rewrite section is a stringified function'?

I'm also not sure what a 'section' is. Is that just the value of the rewrite field?

### PR DESCRIPTION
**Pull request by nkronlage**
This is an attempt to clean up the "Rewrite" section to fix #236. 

Some questions I had during the cleanup:

  - is the rewrite response object's 'query' property really an array?
  - Are the examples at the end correct? The last 2 in particular seem wrong...

### CODE REVIEW

**Comment by wohali** [comment]
Have you validated that the new example actually works?

**Comment by nkronlage** [comment]
Sorry, what do you mean? I did not add any new examples. I changed the spacing to make them easier to read.

**Comment by wohali** [comment]
@nkronlage sorry about that, I looked too quickly and thought you had changed the actual semantics. (It was the variable rename that threw me).

Thank you very much for your contribution to CouchDB!

File: src/api/ddoc/rewrites.rst [hunk 1]
```diff
@@ -21,33 +21,40 @@
...
-      if *code* field provided. If POST/PUT request is being rewritten and no
-      body returned by rewrite function, original request body is used
+    document. The rewrite rules are defined by the ``rewrites`` field of the
+    design document. The ``rewrites`` field can either be a *string* containing
+    the a rewrite function or an *array* of rule definitions.
+
+Using a stringified function for ``rewrites``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    *The 'Rewrite using JS' feature was introduced in CouchDB 1.7*. When the
```
Comment (flimzy) [inline src/api/ddoc/rewrites.rst]: We have a standard 'version added' markup that might make sense for this note.  For example [here](https://github.com/apache/couchdb-documentation/blame/54b7880ee442e5271d695a744ce7d85831a7e9de/src/config/auth.rst#L157).

 I know you didn't change this, so consider this an optional suggestion.

File: src/api/ddoc/rewrites.rst [hunk 2]
```diff
@@ -91,51 +98,58 @@ Rewrite section a is stringified function
...
+    - The *from* pattern and URL are first split on ``/`` to get a list of
+      tokens. For example, if *from* field is ``/somepath/:var/*`` and the URL
+      is ``/somepath/a/b/c``, the tokens are ``somepath``, ``:var``, and
+      ``*`` for the *from* pattern and ``somepath``, ``a``, ``b``, and
+      ``c`` for the URL.
+    - Each token starting with ``:`` in the pattern will match the
+      corresponding token in the URL and define a new dynamic variable whose
+      name is the remaining string after the ``:`` and value is the token from
+      the URL. In this example, the ``:var`` token will match ``b``
+      and set ``var`` = ``a``.
```
Comment (flimzy) [inline src/api/ddoc/rewrites.rst]: This part confuses me. `:var` will match `b`, and set `var` = `a`? Is there a typo here, or is it just confusing logic?  Or is my brain just mush?

**Comment by wohali** [comment]
Hi @nkronlage - to answer one of your questions:

The rewrite response object's `query` is a list of the same format as the `headers` object. [Here's the code that validates the format](https://github.com/apache/couchdb/blob/master/src/couch/src/couch_query_servers.erl#L383-L390), and [here's one example from the test suite](https://github.com/apache/couchdb/blob/master/test/javascript/tests/rewrite_js.js#L51).

The best thing to do with examples is to try them! Do you have the time to do so?

**Comment by wohali** [comment]
This is better than what we have at present, so I'm just going to accept it as-is.

**Pull request merged**